awx/main/tests/unit/utils/test_event_filter.py

```diff
@@ -99,3 +100,46 @@ def test_large_data_payload(fake_callback, fake_cache, wrapped_handle):
     assert recomb_data['role'] == 'some_path_to_role'
     assert 'event' in recomb_data
     assert recomb_data['event'] == 'foo'
+
+
+def test_event_lazy_parsing(fake_callback, fake_cache, wrapped_handle):
+    # Pretend that this is done by the Ansible callback module
+    fake_cache[':1:ev-{}'.format(EXAMPLE_UUID)] = {'event': 'foo'}
+    buff = StringIO()
+    event_data_to_encode = {
+        'uuid': EXAMPLE_UUID,
+        'host': 'localhost',
+        'role': 'some_path_to_role'
+    }
+    write_encoded_event_data(buff, event_data_to_encode)
+
+    # write the data to the event filter in chunks to test lazy event matching
```

Comment:
The purpose of this test is to send the `\x1b[K` we're looking for across multiple `write()` calls to make sure it's properly detected.